src/jsonUtils/computeYogaNode.js

```diff
@@ -0,0 +1,336 @@
+import createStringMeasurer from '../utils/createStringMeasurer';
+import hasAnyDefined from '../utils/hasAnyDefined';
+import pick from '../utils/pick';
+import computeTextTree from './computeTextTree';
+import { INHERITABLE_FONT_STYLES } from '../utils/constants';
+import isNullOrUndefined from '../utils/isNullOrUndefined';
+import { getSymbolMasterByName } from '../symbol';
+
+// flatten all styles (including nested) into one object
+export const getStyles = (node: TreeNode): ViewStyle | Object => {
+  let style = node.props.style;
+
+  if (Array.isArray(style)) {
+    const flattened = Array.prototype.concat.apply([], style);
+    const themeFlattened = Array.prototype.concat.apply([], flattened);
+    const objectsOnly = themeFlattened.filter(f => f);
+    style = Object.assign({}, ...objectsOnly);
+  }
+
+  return style;
+};
+
+const computeYogaNode = (
+  node: TreeNode,
+  context: Context
+): { node: TreeNode, stop?: boolean } => {
+  const yogaNode = yoga.Node.create();
+  const hasStyle = node.props && node.props.style;
+  const style: ViewStyle | Object = hasStyle ? getStyles(node) : {};
+
+  // Setup default symbol instance dimensions
+  if (node.type === 'symbolinstance') {
+    const symbolProps = node.props;
+    const { frame } = getSymbolMasterByName(symbolProps.masterName);
+    yogaNode.setWidth(frame.width);
+    yogaNode.setHeight(frame.height);
+  }
+
+  if (hasStyle) {
+    // http://facebook.github.io/react-native/releases/0.48/docs/layout-props.html
+
+    // Width
+    if (!isNullOrUndefined(style.width)) {
+      yogaNode.setWidth(style.width);
+    }
+
+    // Height
+    if (!isNullOrUndefined(style.height)) {
+      yogaNode.setHeight(style.height);
+    }
+
+    // Min-Height
+    if (!isNullOrUndefined(style.minHeight)) {
+      yogaNode.setMinHeight(style.minHeight);
+    }
+
+    // Min-Width
+    if (!isNullOrUndefined(style.minWidth)) {
+      yogaNode.setMinWidth(style.minWidth);
+    }
+
+    // Max-Height
+    if (!isNullOrUndefined(style.maxHeight)) {
+      yogaNode.setMaxHeight(style.maxHeight);
+    }
+
+    // Min-Width
+    if (!isNullOrUndefined(style.maxWidth)) {
+      yogaNode.setMaxWidth(style.maxWidth);
+    }
+
+    // Margin
+    if (!isNullOrUndefined(style.marginTop)) {
+      yogaNode.setMargin(yoga.EDGE_TOP, style.marginTop);
+    }
+    if (!isNullOrUndefined(style.marginBottom)) {
+      yogaNode.setMargin(yoga.EDGE_BOTTOM, style.marginBottom);
+    }
+    if (!isNullOrUndefined(style.marginLeft)) {
+      yogaNode.setMargin(yoga.EDGE_LEFT, style.marginLeft);
+    }
+    if (!isNullOrUndefined(style.marginRight)) {
+      yogaNode.setMargin(yoga.EDGE_RIGHT, style.marginRight);
+    }
+    if (!isNullOrUndefined(style.marginVertical)) {
+      yogaNode.setMargin(yoga.EDGE_VERTICAL, style.marginVertical);
+    }
+    if (!isNullOrUndefined(style.marginHorizontal)) {
+      yogaNode.setMargin(yoga.EDGE_HORIZONTAL, style.marginHorizontal);
+    }
+    if (!isNullOrUndefined(style.margin)) {
```

Comment:
I'm almost certain you're already correct, but can we check that the order isn't important? I.e. `{ marginTop: 5, margin: 10 }` should have a top margin of 5px.

Reply:
Order shouldn't matter here as I don't call `calculateLayout()` on any of the nodes, until all of their layout styles have already been populated.

Reply:
Didn't think it would matter. Just double checking!

Reply:
👍